### PR TITLE
pkg/emb6: Add random functions to rename patch

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

The `random_init` function from the emb6 package conflicts with the RIOT `random_init` function.
The patch included in the emb6 packages is altered to rename the random functions from the emb6 package by prefixing `emb6_`. Both functions are renamed to keep the package code consistent.

### Issues/PRs references

Solves the build failure from #9535

### CODE REVIEW

File: pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch [hunk 1]
```diff
@@ -1,39 +1,40 @@
-From 4a74cf7e617a7e684dc0a5b5534e26f9d3cb5817 Mon Sep 17 00:00:00 2001
+From 4e6d1640adde2828f10320d39f8996c6e89f82ac Mon Sep 17 00:00:00 2001
 From: Martine Lenders <mail@martine-lenders.eu>
 Date: Thu, 4 Feb 2016 22:15:23 +0100
-Subject: [PATCH 2/4] Rename colliding files and functions
+Subject: [PATCH 2/2] Rename colliding files and functions
```
Comment (miri64) [inline pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch]: :-/ I usually try to `git format-patch` the whole patch collection so this doesn't happen.

Comment (bergzand) [reply pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch]: But patch number 3 got removed at some point

Comment (bergzand) [reply pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch]: Want me to add a commit changing the patches to fix the numbering?

Comment (miri64) [reply pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch]: Ah. Well then forget what I said :D

Comment (miri64) [reply pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch]: > Want me to add a commit changing the patches to fix the numbering?

That's optional, but I wouldn't be mad about it ;-)

Comment (bergzand) [reply pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch]: Done

**Comment by bergzand** [comment]
Tested on two samr21-xpro's on IOT-lab. Package still builds and the udp server/send commands work.

**Comment by bergzand** [comment]
I've reworded and added a line of clarification to the first commit to make it appear less as if I've picked a few random functions to rename.

**Pull request merged**